app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,21 @@
+class VotesController < ApplicationController
+  def create
+    @vendor = Vendor.find(params[:vendor_id])
+    @review = @vendor.reviews.find(params[:review_id])
+
+    if Vote.user_already_voted(current_user, @review.id).empty? == false
+      Vote.destroy_user_vote(current_user, @review)
+      @review = @vendor.reviews.find(params[:review_id])
+    end
+
+    vote = @review.votes.new(value: params[:vote][:vote_value])
+    vote.user = current_user
+
+    if vote.save
+      review = @review
+      review.update_attributes(vote_value: review.vote_value += vote.value)
```

Comment:
As of right now, the vote value will get added into this total when the vote is saved, but it looks like the vote value will not get updated when a vote gets deleted. Make a failing test for this situation, and then fix it! 

Also, I would suggest that you guys consider *not* saving this value in the database, but instead calculating it each time. Each method has its pros and cons - right now, you're saving redundant information in the database (since the info you're saving could be otherwise calculated based on other database information) but you're also needing to make fewer database calls to retrieve this info. It's kind of a matter of opinion which is better in what situation, but I'm generally a fan of the fewest duplicated information possible. Something to think about!